src/test_checker/mod.rs

```diff
@@ -168,7 +153,7 @@ impl SyscallMock for TestChecker {
                 None => panic!("scriptkeeper bug: stderr expected, but not captured"),
                 Some(captured_stderr) => {
                     if &captured_stderr != expected_stderr {
-                        self.register_error(format!(
+                        self.result.register_error(format!(
```

Comment:
This is the place where #201 is actually fixed. `rustc-1.35.0` warned about self being mutably borrowed here (by `register_error`), but also immutably borrowed 5 lines above in `&self.test.stderr`. Since `register_error` was now moved from `TestChecker` to `CheckerResult`, this no longer clashes.

An alternative fix would be to clone `stderr` above. @matthandlersux, @data-pup: Any opinions or other ideas?

Reply:
no opinion, this seems to work just fine